lib/salus/report.rb

```diff
@@ -5,160 +5,109 @@ module Salus
+      scans = @scan_reports.map { |report, _required| [report.scanner_name, report.to_h] }.to_h
+
       {
+        version: VERSION,
         project_name: @project_name,
-        scans: @scans,
-        info: @info,
+        passed: passed?,
+        scans: scans,
         errors: @errors,
-        version: VERSION,
         custom_info: @custom_info,
-        configuration: @configuration
-      }
-    end
-
-    def to_json
-      JSON.pretty_generate(to_h)
+        config: @config
+      }.compact
     end
 
     # Generates the text report.
-    def to_s(verbose: false)
-      lines = []
-      lines << "#{SPECIAL_BAR} Salus Scan v#{VERSION} for #{@project_name} #{SPECIAL_BAR}"
-
-      if @scans.any?
-        @scans.each do |scanner, scan_data|
-          # Scanner title which gives the high level picture
-          scanner_title = "\t#{scanner}"
-          scan_result = SCAN_RESULT_WORD[scan_data['passed']]
-          scanner_title += " => #{scan_result}" unless scan_result.nil?
-          lines << scanner_title
-
-          # Additional data about the scan. Give stdout, stderr and if verbose, also give info.
-          lines << "STDOUT:\n#{scan_data['stdout']}" unless scan_data['stdout'].nil?
-          lines << "STDERR:\n#{scan_data['stderr']}" unless scan_data['stderr'].nil?
-
-          if verbose
-            scan_data['info']&.each do |type, messages|
-              lines << "INFO - #{type}"
-              lines += messages.map { |message| "\t#{message}" }
-            end
-          end
-        end
+    def to_s(verbose: false, wrap: WRAP)
+      output = "==== Salus Scan v#{VERSION}"
+      output += " for #{@project_name}" unless @project_name.nil?
+
+      # Sort scan reports required before optional, failed before passed,
+      # and alphabetically by scanner name
+      scan_reports = @scan_reports.sort_by do |report, required|
+        [
+          required ? 0 : 1,
+          report.passed? ? 1 : 0,
+          report.scanner_name
+        ]
       end
 
-      if verbose && @info.any?
-        lines << "#{BAR} Scan Info #{BAR}"
-        @info.each do |type, messages|
-          lines << type
-          lines += messages.each { |message| "\t#{message}" }
-        end
+      # Print a summary of the scan results
+      # FIXME(as3richa): this should be a pretty table :)
+      output += "\n\nRan #{scan_reports.length} scanner(s):"
+      scan_reports.each do |report, required|
+        output += "\n- #{report.scanner_name}: #{report.passed? ? 'PASSED' : 'FAILED'}"
+        output += " in #{report.running_time}s" unless report.running_time.nil?
+        output += required ? ' (required)' : ' (not required)'
+      end
+
+      scan_reports.each do |report, _required|
+        output += "\n\n#{report.to_s(verbose: verbose, wrap: wrap)}"
       end
 
-      # Only add configuration if verbose mode is on.
+      # Adjust the wrap by 2 because we're wrapping indented paragraphs
+      wrap = (wrap.nil? ? nil : wrap - 2)
```

Comment:
nit: make this a new var instead of overwriting. Might stop a bug in the future if someone messes with this method.